cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -312,4 +308,52 @@ public void removeProfileAssignment(ProfileId profileId, EntityId entityId) thro
   private ProfileDataset getProfileDataset(DatasetContext context) {
     return ProfileDataset.get(context, datasetFramework);
   }
+
+  private void deleteProfile(DatasetContext context, ProfileId profileId,
```

Comment:
should probably refactor this to take the ProfileDataset and AppMetadataStore to avoid getting it every time.